src/tensorflow.cc

```diff
@@ -1280,31 +1273,36 @@ AutoCompleteHelper::FixBatchingSupport()
               } else {
                 config_io.MemberAsArray("dims", &config_dims);
               }
-              if (config_dims.ArraySize() != 0) {
-                // look up corresponding io info from model
-                for (const TRITONTF_IOList* itr = model_ios[ios_idx];
-                     itr != nullptr; itr = itr->next_) {
-                  TRITONTF_IO* io = itr->io_;
-                  if (config_name == io->name_) {
-                    bool should_batch =
-                        (io->shape_->rank_ == (config_dims.ArraySize() + 1));
-                    // inconsistent hint
-                    if (config_batch_hint &&
-                        (model_support_batching_ != should_batch)) {
-                      return TRITONSERVER_ErrorNew(
-                          TRITONSERVER_ERROR_INTERNAL,
-                          std::string(
-                              "unable to autofill for '" +
-                              model_state_->Name() +
-                              "', model tensor configurations are "
-                              "contradicting " +
-                              "each other in terms of whether batching is "
-                              "supported")
-                              .c_str());
-                    }
-                    config_batch_hint = true;
-                    model_support_batching_ = should_batch;
+              for (const TRITONTF_IOList* itr = model_ios[ios_idx];
+                   itr != nullptr; itr = itr->next_) {
+                TRITONTF_IO* io = itr->io_;
+                if (config_name == io->name_) {
+                  bool model_io_explicit = io->shape_->rank_ > 0;
+                  bool model_shape_matches_config_shape =
+                      io->shape_->rank_ == config_dims.ArraySize() + 1
+
+                                               // Check for false hints
+                                               if (!model_io_explicit &&
+                                                   config_dims.ArraySize() == 0)
+                  {
+                    // don't have enough information to batch
+                    config_batch_hint = false;
+                  }
+                  else if (
+                      model_io_explicit && !model_shape_matches_config_shape)
```

Comment:
Wouldn't this throw error if `model_io_explicit == true`
and the model shapes exactly matches the config shape?

```
model dims: [ -1, -1 ]
config dims: [ 2, 2 ]
```

Reply:
This is an interesting case. The model says it is able to batch 2x2 tensors but the configuration specifies a single subset of the model's space. So while the configuration is correct, there is an implicit -1 added prepended to the config dims, `[ -1, 2, 2]`.  If we want to batch then according to the model config we can only batch 2x2 tensors. The model config states it can batch any number of 1xN tensors. For your given scenario, batching would have to be turned off for the config to be correct.

Reply:
I think the right choice here is to disable the batching support.
Assuming non-ragged input/outputs, if the user provided dimension counts exactly matches the model dims, which is 2 in this case. This means Triton should not enable batching. It should be treated as a hint that max_batch_size should be left 0.